diff --git a/trunk/src/plugins/wscript b/trunk/src/plugins/wscript
index 49cde75e..f969b74f 100644
--- a/trunk/src/plugins/wscript
+++ b/trunk/src/plugins/wscript
@@ -164,4 +164,4 @@ def build(bld):
         tsk.env['shlib_PATTERN'] = '%s.so'
 
 def configure(conf):
-    conf.check_cfg(package='eigen3', args='--cflags', uselib_store="EIGEN3", mandatory=1)
+    pass
diff --git a/trunk/wscript b/trunk/wscript
index d2a838fe..5ecf9f1c 100644
--- a/trunk/wscript
+++ b/trunk/wscript
@@ -833,6 +833,8 @@ def configure(conf):
             sub_file('./libgxwmm/gxwmm.pc.in', './libgxwmm/gxwmm.pc', (('prefix=/path', 'prefix=%s' % pa), ))
             sub_file('./libgxwmm/gxwmm.pc', './libgxwmm/gxwmm.pc', (('libdir=/path','libdir=%s' % conf.env['LIBDIR']), ))
 
+    conf.env['CXXFLAGS'] += ["-I%s" % os.getenv("EIGEN3_PATH")]
+
     if opt.jobs:
         conf.env['JOBS'] = str(opt.jobs)
 
