diff --git a/trunk/tools/dsp2cc b/trunk/tools/dsp2cc
index 958ea690..05352f37 100755
--- a/trunk/tools/dsp2cc
+++ b/trunk/tools/dsp2cc
@@ -505,7 +505,7 @@ class Parser(object):
         s = {}
         self.ui = UIDefs()
         #self.readMeta()  # (needed only for faust 9.4
-        self.headvers = self.skip_until(r"\s*//\s*(Code generated with Faust.*)").group(1)
+        self.headvers = self.skip_until(r"\s*(Code generated with Faust.*)").group(1)
         s["includes"] = self.readIncludes(r"  private:")
         #self.skip_until(r"  private:")
         var_decl = self.copy(r"  public:")
diff --git a/trunk/src/LV2/faust/gx_vibrochump.dsp b/trunk/src/LV2/faust/gx_vibrochump.dsp
index 88f77167..6d9d78db 100644
--- a/trunk/src/LV2/faust/gx_vibrochump.dsp
+++ b/trunk/src/LV2/faust/gx_vibrochump.dsp
@@ -25,6 +25,30 @@ import("redeye.lib");
 **	12AX7:EQ:(12AX7:6V6)~FEEDBACK 
 */
 
+bmptone = pre : iir((b0/a0,b1/a0,b2/a0),(a1/a0,a2/a0)) with {
+    LogPot(a, x) = if(a, (exp(a * x) - 1) / (exp(a) - 1), x);
+    Inverted(b, x) = if(b, 1 - x, x);
+    s = 0.993;
+    fs = float(SR);
+    pre = _;
+
+    
+        
+        Tone = vslider("Tone[name:Tone]", 0.5, 0, 1, 0.01) : Inverted(0) : smooth(s);
+            
+    b0 = Tone*(3.68688858465455e-10*pow(fs,2) - 0.00316091270975185) + 8.67101574539126e-7*fs + 0.00401435914138484;
+
+    b1 = Tone*(-7.37377716930911e-10*pow(fs,2) - 0.00632182541950369) + 0.00802871828276969;
+
+    b2 = Tone*(3.68688858465455e-10*pow(fs,2) - 0.00316091270975185) - 8.67101574539126e-7*fs + 0.00401435914138484;
+
+    a0 = fs*(3.68688858465455e-10*fs + 3.03485551088694e-6) + 0.00486780557301784;
+
+    a1 = -7.37377716930911e-10*pow(fs,2) + 0.00973561114603569;
+
+    a2 = fs*(3.68688858465455e-10*fs - 3.03485551088694e-6) + 0.00486780557301784;
+};
+
 /*
 ** Model of a vactrol tremolo unit by "transmogrify"
 ** c.f. http://sourceforge.net/apps/phpbb/guitarix/viewtopic.php?f=7&t=44&p=233&hilit=transmogrifox#p233
diff --git a/trunk/src/LV2/gx_redeye.lv2/gx_redeye.ttl b/trunk/src/LV2/gx_redeye.lv2/gx_redeye.ttl
index e73dd3f1..1339152d 100644
--- a/trunk/src/LV2/gx_redeye.lv2/gx_redeye.ttl
+++ b/trunk/src/LV2/gx_redeye.lv2/gx_redeye.ttl
@@ -181,9 +181,9 @@ rdfs:comment """
         lv2:index 0 ;
         lv2:symbol "Gain" ;
         lv2:name "Gain";
-        lv2:default 0.5 ;
-        lv2:minimum 0.0 ;
-        lv2:maximum 1.0 ;
+        lv2:default 5.0 ;
+        lv2:minimum -20.0 ;
+        lv2:maximum 10.0 ;
     ] , [
         a lv2:InputPort ,
             lv2:ControlPort ;
@@ -288,9 +288,9 @@ rdfs:comment """
         lv2:index 0 ;
         lv2:symbol "Gain" ;
         lv2:name "Gain";
-        lv2:default 0.5 ;
-        lv2:minimum 0.0 ;
-        lv2:maximum 1.0 ;
+        lv2:default 5.0 ;
+        lv2:minimum -20.0 ;
+        lv2:maximum 10.0 ;
     ] , [
         a lv2:InputPort ,
             lv2:ControlPort ;
