diff --git a/trunk/src/LV2/gx_jcm800pre.lv2/jcm800pre.cc b/trunk/src/LV2/gx_jcm800pre.lv2/jcm800pre.cc
index 9415467e..c3cca5df 100644
--- a/trunk/src/LV2/gx_jcm800pre.lv2/jcm800pre.cc
+++ b/trunk/src/LV2/gx_jcm800pre.lv2/jcm800pre.cc
@@ -3836,7 +3836,7 @@ typedef enum
 
 void DKPlugin::init(unsigned int samplingFreq, PluginLV2 *plugin) {
     DKPlugin& self = *static_cast<DKPlugin*>(plugin);
-    self.smp.setup(samplingFreq, 96000);
+    self.smp.setup(samplingFreq, samplingFreq);
     self.X.setZero();
    self.Y.setZero();
    unsigned int fs = samplingFreq;
diff --git a/trunk/src/LV2/gx_jcm800pre_st.lv2/jcm800pre.cc b/trunk/src/LV2/gx_jcm800pre_st.lv2/jcm800pre.cc
index 9415467e..c3cca5df 100644
--- a/trunk/src/LV2/gx_jcm800pre_st.lv2/jcm800pre.cc
+++ b/trunk/src/LV2/gx_jcm800pre_st.lv2/jcm800pre.cc
@@ -3836,7 +3836,7 @@ typedef enum
 
 void DKPlugin::init(unsigned int samplingFreq, PluginLV2 *plugin) {
     DKPlugin& self = *static_cast<DKPlugin*>(plugin);
-    self.smp.setup(samplingFreq, 96000);
+    self.smp.setup(samplingFreq, samplingFreq);
     self.X.setZero();
    self.Y.setZero();
    unsigned int fs = samplingFreq;
