diff --git a/trunk/src/LV2/gx_amp_stereo.lv2/gxamp_stereo.cpp b/trunk/src/LV2/gx_amp_stereo.lv2/gxamp_stereo.cpp
index 144dafce..8a90411b 100644
--- a/trunk/src/LV2/gx_amp_stereo.lv2/gxamp_stereo.cpp
+++ b/trunk/src/LV2/gx_amp_stereo.lv2/gxamp_stereo.cpp
@@ -247,7 +247,7 @@ private:
   void update_pre()
     {pre = (alevel_);}
   inline bool val_changed() 
-    {return  std::abs(alevel_ - (*alevel)) > 0.1 || abs(clevel_ - (*clevel)) > 0.1 || std::abs(c_model_ - (*c_model)) > 0.1;}
+    {return std::abs(alevel_ - (*alevel)) > 0.1 || std::abs(clevel_ - (*clevel)) > 0.1 || std::abs(c_model_ - (*c_model)) > 0.1;}
 
   // LV2 stuff
   LV2_URID_Map*                map;
